tests/virt_autotest/setup_dns_service.pm

```diff
@@ -0,0 +1,223 @@
+# SUSE's openQA tests
+#
+# Copyright ?? 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Setup dns service for virtual machines to have dns compatiable name.
+# Maintainer: Wayne Chen <wchen@suse.com>
+
+use base "virt_autotest_base";
+use strict;
+use warnings;
+use testapi;
+
+sub run {
+    my $self = shift;
+
+    my $wait_script          = "30";
+    my $bridgeif             = "br123";
+    my $bridgeip             = "192.168.123.1";
+    my $dns_domain_forward   = "testvirt.net";
+    my $dns_domain_reverse   = "123.168.192";
+    my $dns_resolver_file    = "/etc/resolv.conf";
+    my $dns_resolver_tmpfile = "/etc/resolv.tmp";
+    my $dhcpd_config_file    = "/etc/dhcpd.conf";
+    my $dhcpd_service_name   = "dhcpd";
+    my $dhcpd_lease_file     = "/var/lib/dhcp/db/dhcpd.leases";
+    my $dhcpd_lease_time     = "28800";
+
+    assert_script_run("zypper --non-interactive in lsb-release bridge-utils bind bind-chrootenv bind-utils");
+
+    my $get_bridgeif_count = "brctl show | grep $bridgeif | wc -l";
+    my $bridgeif_count     = script_output($get_bridgeif_count, $wait_script, type_command => 0, proceed_on_failure => 1);
+    if ($bridgeif_count eq '0') {
+        assert_script_run("ip link add name $bridgeif type bridge");
+        assert_script_run("ip addr add $bridgeip/24 dev $bridgeif");
+        assert_script_run("ip link set dev $bridgeif up");
+    }
+
+    #Add $bridgeip as additional nameserver
+    #Add $dns_domain_forward as additional search domain
+    my $add_resolver_nameserver = "awk \'FNR==NR{ if (/^nameserver/) p=NR\; next} 1; FNR==p{ print \"nameserver $bridgeip\" }\' $dns_resolver_file $dns_resolver_file > $dns_resolver_tmpfile";
+    assert_script_run("cp $dns_resolver_file $dns_resolver_file.orig");
+    assert_script_run("$add_resolver_nameserver");
+    assert_script_run("mv $dns_resolver_tmpfile $dns_resolver_file");
+    my $add_resolver_search = "sed -irn \'s/^search.*\$/& $dns_domain_forward/\' $dns_resolver_file";
+    assert_script_run("$add_resolver_search");
+
+    #Reset default-lease-time and max-lease-time in /etc/dhcpd.conf to 28800
+    my $set_dhcpd_lease_time = "sed -ir \"s/default-lease-time\.*\$/default-lease-time $dhcpd_lease_time;/g; s/max-lease-time.*\$/max-lease-time $dhcpd_lease_time;/g;\" $dhcpd_config_file";
+    assert_script_run("$set_dhcpd_lease_time");
+
+    my $get_os_installed_release = "lsb_release -r | grep -oE \"[[:digit:]]{2}\"";
+    my $os_installed_release     = script_output($get_os_installed_release, $wait_script, type_command => 0, proceed_on_failure => 1);
```

Comment:
Does this script run on the host of the existing test suites in openQA? If yes, the test settings, VERSION_TO_INSTALL VERSION may tell the SLE release installed on host.

Reply:
This module usually runs after guest installation run